### ISSUE TITLE
Dynamic support broken

### ISSUE THREAD

**Issue by MgSam** [RSuter/NSwag#1341]
Related to https://github.com/RSuter/NSwag/issues/1216. The fix that removed `ExpandoObject` and used `any` instead worked for most of our cases, but surfaced another bug. 

When generating TypeScript for a Web API method that accepts a `dynamic`, `ExpandoObject, or `object` parameter, the wrong code is generated.

It is generating this:
``` TypeScript
foo(bar: any | null): Observable<void> {
        let url_ = this.baseUrl + "/api/Baz/Foo?";
        if (bar === undefined)
            throw new Error("The parameter 'bar' must be defined.");
        else
            url_ += "bar=" + encodeURIComponent("" + bar) + "&"; 
...
}
```

Instead of this:
``` TypeScript
foo(bar: any | null): Observable<void> {
        let url_ = this.baseUrl + "/api/Baz/Foo";
        url_ = url_.replace(/[?&]$/, "");
 
        const content_ = JSON.stringify(bar);
...
```

In other words, its treating `dynamic`, `ExpandoObject`, or `object` parameters the same as it treats primitives. It instead should be calling `JSON.stringify` on them.

**Comment by RSuter** [comment]
It's probably treated as primitive parameter and not body parameter, adding [FromBody] should work but maybe we should handle it this way by default?

**Comment by MgSam** [comment]
Thanks! `[FromBody]` worked. 

I agree it should handle it this way by default. Maybe the behavior should be that for an unknown type `stringify` is the default behavior and the only certain built-in primitive types are special cased to get the URI encoding (int, string, etc).

**Comment by RSuter** [comment]
This is a problem of the Swagger generator and not the code generator: Withou the FromBody attribute the parameter is described as query param and the code is “correctish”...

Is the object param handled as body param without FromBody by asp.net?

**Comment by steji113** [comment]
I wonder if this is due to the new ASP.NET Core 2.1 handling of complex types being inferred automatically: https://blogs.msdn.microsoft.com/webdev/2018/02/27/asp-net-core-2-1-web-apis/

**Comment by RSuter** [comment]
Yes, it's possible that this has changed in asp.net core 2.1 - there we have a lot more predefined conventions...

**Comment by RSuter** [comment]
We are now working on the new API Explorer based generator (its almost ready) where conventions like that are automatically processed, eventually we will only support this new generator. Use the UseSwagger*WithApiExplorer() methods to use it now...
